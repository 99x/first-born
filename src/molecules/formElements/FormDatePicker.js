import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { DatePicker } from "../../atoms/DatePicker";
import { Text } from "../../atoms/Text";

export class FormDatePicker extends Component {

    render() {
        const { label, ...otherProps } = this.props;

        return (
            <View style={styles.container}>
                <Text size="sub_heading" color="rgba(33, 33, 33, 0.87)">{label}</Text>
                <DatePicker {...otherProps} />
            </View>
        )
    }
}

FormDatePicker.propTypes = {
    label: PropTypes.string.isRequired,
    ...DatePicker.propTypes
}

const styles = StyleSheet.create({
    container: {
        width: "100%"
    }
})