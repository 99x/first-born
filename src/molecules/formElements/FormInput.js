import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { Input } from "../../atoms/Input";
import { Text } from "../../atoms/Text";
import { commonColors } from "../../utils/color";

export class FormInput extends Component {
    render() {
        const { label, ...otherProps } = this.props;

        return (
            <View style={styles.container}>
                <Text size="sub_heading" color={commonColors.darkGrey}>
                    {label}
                </Text>
                <Input {...otherProps} />
            </View>
        );
    }
}

FormInput.propTypes = {
    label: PropTypes.string.isRequired,
    ...Input.propTypes
};

const styles = StyleSheet.create({
    container: {
        width: "100%"
    }
});
