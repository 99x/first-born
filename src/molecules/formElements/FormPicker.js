import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { Picker } from "../../atoms/Picker";
import { Text } from "../../atoms/Text";
import { commonColors } from "../../utils/color";

export class FormPicker extends Component {
    render() {
        const { label, children, data, ...otherProps } = this.props;

        return (
            <View style={styles.container}>
                <Text size="sub_heading" color={commonColors.darkGrey}>
                    {label}
                </Text>
                <Picker {...otherProps}>
                    {children
                        ? children
                        : data
                        ? data.map((dataElement, key) => (
                              <Picker.Item {...dataElement} key={key} />
                          ))
                        : null}
                </Picker>
            </View>
        );
    }
}

FormPicker.propTypes = {
    label: PropTypes.string.isRequired,
    date: PropTypes.array,
    ...Picker.propTypes
};

const styles = StyleSheet.create({
    container: {
        width: "100%"
    }
});
