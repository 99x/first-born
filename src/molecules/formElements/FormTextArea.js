import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { TextArea } from "../../atoms/TextArea";
import { Text } from "../../atoms/Text";
import { commonColors } from "../../utils/color";

export class FormTextArea extends Component {

    render() {
        const { label, ...otherProps } = this.props;

        return (
            <View style={styles.container}>
                <Text size="sub_heading" color={commonColors.darkGrey}>{label}</Text>
                <TextArea {...otherProps} />
            </View>
        )
    }
}

FormTextArea.propTypes = {
    label: PropTypes.string.isRequired,
    ...TextArea.propTypes
}

const styles = StyleSheet.create({
    container: {
        width: "100%"
    }
})