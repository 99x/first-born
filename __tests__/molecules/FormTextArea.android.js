import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { FormTextArea } from "../../src/molecules/formElements/FormTextArea";

jest.mock("Platform", () => {
    const Platform = require.requireActual("Platform");
    Platform.OS = "android";
    return Platform;
});

it("renders regular FormTextArea", () => {
    const tree = renderer.create(<FormTextArea label="Address" />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders colored FormTextArea", () => {
    const tree = renderer
        .create(<FormTextArea label="Address" color="red" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders FormTextArea with placeholder", () => {
    const tree = renderer
        .create(<FormTextArea label="Address" placeholder="placeholder" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
