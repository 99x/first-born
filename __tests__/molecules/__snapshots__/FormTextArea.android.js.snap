// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FormTextArea with placeholder 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgba(33, 33, 33, 0.87)",
        "fontSize": 15,
        "fontWeight": "normal",
        "marginVertical": 5,
        "textAlign": "left",
      }
    }
  >
    Address
  </Text>
  <View
    style={
      Object {
        "borderColor": "rgba(33, 33, 33, 0.5)",
        "borderRadius": 10,
        "borderWidth": 0.9,
        "marginVertical": 10,
        "paddingHorizontal": 5,
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onBlur={[Function]}
      onContentSizeChange={[Function]}
      onFocus={[Function]}
      placeholder="placeholder"
      style={
        Object {
          "height": 100,
          "textAlignVertical": "top",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`renders colored FormTextArea 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgba(33, 33, 33, 0.87)",
        "fontSize": 15,
        "fontWeight": "normal",
        "marginVertical": 5,
        "textAlign": "left",
      }
    }
  >
    Address
  </Text>
  <View
    style={
      Object {
        "borderColor": "rgba(33, 33, 33, 0.5)",
        "borderRadius": 10,
        "borderWidth": 0.9,
        "marginVertical": 10,
        "paddingHorizontal": 5,
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onBlur={[Function]}
      onContentSizeChange={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "height": 100,
          "textAlignVertical": "top",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`renders regular FormTextArea 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgba(33, 33, 33, 0.87)",
        "fontSize": 15,
        "fontWeight": "normal",
        "marginVertical": 5,
        "textAlign": "left",
      }
    }
  >
    Address
  </Text>
  <View
    style={
      Object {
        "borderColor": "rgba(33, 33, 33, 0.5)",
        "borderRadius": 10,
        "borderWidth": 0.9,
        "marginVertical": 10,
        "paddingHorizontal": 5,
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onBlur={[Function]}
      onContentSizeChange={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "height": 100,
          "textAlignVertical": "top",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
